protocols/syslog/syslog.c

```diff
@@ -34,88 +35,142 @@
 #include "syslog_net.h"
 
 
-static char send_buffer[MAX_DYNAMIC_SYSLOG_BUFFER + 1];
+static char send_buffer[MAX_LINES_SYSLOG_BUFFER][MAX_LENGTH_SYSLOG_BUFFER + 1];
 extern uip_udp_conn_t *syslog_conn;
 static struct SyslogCallbackCtx syslog_callbacks[SYSLOG_CALLBACKS];
```

Comment:
Does it make sense to hold the send_buffer and the callback pointer in a structure, that is dynamicall allocated? You still have to remember the allocated structures. I suggest a single linked list with a global variable that points to the first entry in the list.